lib/Froxlor/Cron/Dns/Bindmirror.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+/**
+ * This file is part of the Froxlor project.
+ * Copyright (c) 2010 the Froxlor Team (see authors).
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, you can also view it online at
+ * https://files.froxlor.org/misc/COPYING.txt
+ *
+ * @copyright  the authors
+ * @author     Froxlor team <team@froxlor.org>
+ * @license    https://files.froxlor.org/misc/COPYING.txt GPLv2
+ */
+
+namespace Froxlor\Cron\Dns;
+
+use Froxlor\Dns\Dns;
+use Froxlor\FileDir;
+use Froxlor\FroxlorLogger;
+use Froxlor\Settings;
+
+class Bindmirror extends DnsBase
+{
+
+	private $bindconf_file = "";
+
+	public function writeConfigs()
+	{
+		// tell the world what we are doing
+		$this->logger->logAction(FroxlorLogger::CRON_ACTION, LOG_INFO, 'Task4 started - Rebuilding froxlor_bind.mirror.conf');
+
+		$domains = $this->getDomainList();
+
+		if (empty($domains)) {
+			$this->logger->logAction(FroxlorLogger::CRON_ACTION, LOG_INFO, 'No domains found for nameserver-config, skipping...');
+			return;
+		}
+
+		$this->bindconf_file = '# ' . Settings::Get('system.bindmirrorconf_directory') . 'froxlor_bind.mirror.conf' . "\n" . '# Created ' . date('d.m.Y H:i') . "\n" . '# Do NOT manually edit this file, all changes will be deleted after the next domain change at the panel.' . "\n";
+		$bindmasterservers = explode(',', Settings::Get('system.bindmasterservers'));
+		if (!empty($bindmasterservers)) {
```

Comment:
this by design cannot be empty at this stage as this class only gets called when it's none-empty (**[lib/Froxlor/Cron/Dns/Bind.php](https://github.com/Froxlor/Froxlor/pull/1107/files#diff-a2a22d06c0ee6dce9d44284969ef12913e03a1658b7d5b56670cdbab89c33d6a)**)